src/HttpMessage.php

```diff
@@ -127,26 +127,19 @@ protected function replaceHeaders(array $headers): void
      *
      * @throws \Error If the header name or value is invalid.
      */
-    protected function setHeader(string $name, array|string $value): void
+    protected function setHeader(string $name, array|string|int|float $value): void
     {
         \assert($this->isNameValid($name), "Invalid header name");
 
         $lcName = HEADER_LOWER[$name] ?? \strtolower($name);
 
-        if (!\is_array($value)) {
```

Comment:
These branches existed because setting headers is part of the highly optimized code path for http-server. These changes have a noticeable impact on performance (~5% less req/sec).